helper/resource/state.go

```diff
@@ -242,6 +247,9 @@ func (conf *StateChangeConf) WaitForState() (interface{}, error) {
 					// target state not reached, save the result for the
 					// TimeoutError and wait for the channel to close
 					lastResult = r
+				case <-ctx.Done():
+					log.Println("[WARN] Context cancelation detected, abandoning grace period")
```

Comment:
I think this should be an `[ERROR]` not a `[WARN]` here as its going to return a `TimeoutError`